bin/transform

```diff
@@ -97,47 +97,61 @@ def parse_geographic_columns(gisaid_data: pd.DataFrame) -> pd.DataFrame:
     Expands the string found in the column named `location` in the given
     *df*, creating four new columns. Returns the modified ``pd.DataFrame``.
     """
-    def titlecase_non_whitelisted_tokens(tokens: Union[List[str],
-        pd._libs.missing.NAType]) -> Optional[str]:
+    def titlecase(text: Union[str, pd._libs.missing.NAType],
+        articles: Set[str] = {}, abbrev: Set[str] = {}) -> Optional[str]:
         """
         Returns a title cased location name from the given location name
-        *tokens*. Ensures that no tokens contained in the *whitelist_tokens* or
-        matching the *whitelist_regex* pattern(s) are converted to title case.
+        *tokens*. Ensures that no tokens contained in the *whitelist_tokens* are
+        converted to title case.
+
+        >>> articles = {'a', 'and', 'of', 'the', 'le'}
+        >>> abbrev = {'USA', 'DC'}
+
+        >>> titlecase("the night OF THE LIVING DEAD", articles)
+        'The Night of the Living Dead'
+
+        >>> titlecase("BRAINE-LE-COMTE, FRANCE", articles)
+        'Braine-le-Comte, France'
+
+        >>> titlecase("auvergne-RHÔNE-alpes", articles)
+        'Auvergne-Rhône-Alpes'
+
+        >>> titlecase("washington DC, usa", articles, abbrev)
+        'Washington DC, USA'
```

Comment:
Does it make sense to keep the doc tests here? They fail when I run `python3 -m doctest -v transform` locally.

Reply:
Would be good to keep the doc tests so we can check them for any future changes.

But ugh. This error is because doctest assumes that files which don't end in `.py` are doctests embedded in text, not code, files:

> Because the file name does not end with .py, doctest infers that it must be run with testfile(), not testmod().

From the end of [Simple usage: checking examples in a text file](https://docs.python.org/3/library/doctest.html#simple-usage-checking-examples-in-a-text-file).

I think an easy fix here is to move `titlecase` into the new lib/… module once #55 is merged (soon).